### ISSUE TITLE
Latest agent version does not mitigate Log4j vulnerability 

### ISSUE THREAD

**Issue by jenle-chime** [DataDog/datadog-agent#10221]
According to Datadog's official announcement, released versions 6.32.2 and 7.32.2 of the Agent that initializes the JVM for our JMX monitoring client with the following flag: `-Dlog4j2.formatMsgNoLookups=True`

A new CVE has been released since: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-45046

Note that previous mitigations involving configuration such as to set the system property `log4j2.noFormatMsgLookup` to `true` do NOT mitigate this specific vulnerability. Log4j 2.16.0 fixes this issue by removing support for message lookup patterns and disabling JNDI functionality by default. This issue can be mitigated in prior releases (<2.16.0) by removing the JndiLookup class from the classpath (example: `zip -q -d log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class`).

**Comment by jduan-highnote** [comment]
Yes, can this to be addressed asap? Thanks!

**Comment by sgnn7** [comment]
Hi @jenle-chime / @jduan-highnote!
We are trying to coalesce/unify all the information together about this together so all official communication updates around log4j will be available at https://www.datadoghq.com/log4j-vulnerability/ as they become available.

I'll leave the issue open for visibility for now but keep in mind that most up-to-date info will be provided in the mentioned link.

**Comment by jenle-chime** [comment]
@sgnn7 I am trying to report an issue with Datadog's latest update. Is there another place you would like us to report log4j issues?

**Comment by knowshan** [comment]
Datadog agent is downloading jmxfetch.jar file using [jmxfetch.rb script](https://github.com/DataDog/datadog-agent/blob/main/omnibus/config/software/jmxfetch.rb ) with versions specified in [release.json](https://github.com/DataDog/datadog-agent/blob/main/release.json). It seems like jmxfetch version uses log4j version v2.12.2 even in the latest release.  IMO, until jmxfetch is updated with log4j v2.16 and Datadog agent uses it, this issue won't be addressed.

Is there any ETA on getting this fixed in jmxfetch and datadog-agent image with jmx?

**Comment by sudosoul** [comment]
They won't upgrade past 2.12.2 because anything past that doesn't support Java 7. 
Where does the log4j jar get downloaded to? Can we just modify the jmxfetch / install script to run 
`zip -q -d log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class`

**Comment by sgnn7** [comment]
>  It seems like jmxfetch version uses log4j version v2.12.2 even in the latest release. IMO, until jmxfetch is updated with log4j v2.16 and Datadog agent uses it, this issue won't be addressed.

@knowshan / @sudosoul I can't comment on the timelines but per the upstream page at https://logging.apache.org/log4j/2.x/, 2.12.2 is not vulnerable:

> The Log4j team has been made aware of a security vulnerability, CVE-2021-45046, that has been addressed in Log4j 2.12.2 for Java 7 and 2.16.0 for Java 8 and up.
> ...
> In version 2.12.2 Log4j disables access to JNDI by default. Usage of JNDI in configuration now need to be enabled explicitly. Calls to the JndiLookup will now return a constant string. Also, Log4j now limits the protocols by default to only java. The message lookups feature has been completely removed.

**Comment by sudosoul** [comment]
@sgnn7  You're  right, thank you - spoke too soon. 
This issue can be closed then.
@jenle-chime

**Comment by kaarolch** [comment]
The question is how datadog pull plugin. If I download plugin 2 days ago (after 7.32.2 was release) and jmxfetch was bump 11 hours ago, should I reload datadaog or need any other action to refresh jmx plugin version?

**Comment by szibis** [comment]
I check that and agent  7.32.2 was installed on 11.Dec using 0.44.4 version of JMXFetch and. changes for using Log4j 2.12.2 are made in 0.44.5 https://github.com/DataDog/jmxfetch/releases/tag/0.44.5
Anyone can confirm that Datadog-agent still has Log4j issue ??

**Comment by amitfred** [comment]
I see version used is 6.26.0 for our service. How to upgrade the version here and as @szibis  said 2.12.2 is still not added to 6.32.2 . So how to proceed and looks like adding only -Dlog4j2.formatMsgNoLookups=True will not work completely as per below link - https://github.com/advisories/GHSA-7rjr-3q55-vv33

Any suggestions here?

**Comment by szibis** [comment]
We are waiting for 7.32.3 and 6.32.3 https://github.com/DataDog/datadog-agent/pull/10240

**Comment by olivielpeau** [comment]
We are currently working on releasing versions 6.32.3 and 7.32.3 of the Agent that include a patched version of the log4j library. We'll update this issue when the release is available.

**Comment by jithumkunjuraman** [comment]
https://github.com/DataDog/datadog-agent/releases/tag/7.32.3

@jenle-chime

**Comment by olivielpeau** [comment]
We have released `7.32.3` and `6.32.3` (https://github.com/DataDog/datadog-agent/releases/tag/7.32.3). These Agent versions include log4j 2.12.2, which fixes both CVE-2021-44228 and CVE-2021-45046, as documented on the [log4j security page](https://logging.apache.org/log4j/2.x/security.html).

Please refer to https://www.datadoghq.com/log4j-vulnerability/ for further information, thanks!

**Comment by nwaldropEXR** [comment]
Will the chef/kitchen recipe be updated with this new package?
https://supermarket.chef.io/cookbooks/datadog#changelog

**Comment by olivielpeau** [comment]
@nwaldropEXR To be clear, our official chef cookbook is not affected by these vulnerabilities. The agent versions that it installs can be affected depending on the agent versions you've configured the cookbook to install. You can pin the version of the Agent that the cookbook installs with the `["datadog"]["agent_version"]` attribute. You'll find more details in the cookbook's [README](https://supermarket.chef.io/cookbooks/datadog#readme)

Please reach out to our [support team](https://www.datadoghq.com/support/) if you have further questions about the chef cookbook, thanks!

**Comment by JorgenG** [comment]
As mentioned in https://github.com/DataDog/datadog-agent/issues/10301 there is yet another CVE which is fixed in 2.12.3.

See: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-45105

> Apache Log4j2 versions 2.0-alpha1 through 2.16.0 (excluding 2.12.3) did not protect from uncontrolled recursion from self-referential lookups. This allows an attacker with control over Thread Context Map data to cause a denial of service when a crafted string is interpreted. This issue was fixed in Log4j 2.17.0 and 2.12.3.

Could you reopen any of these issues to track it until finished?

**Comment by sgnn7** [comment]
Hey @JorgenG,
 https://www.datadoghq.com/log4j-vulnerability/ has already been updated with the infromation about that CVE:
```
Versions 6.32.3 & 7.32.3:
...
Our analysis of CVE-2021-45105 indicates that the JMXfetch client is not affected. The implementation
has a hardcoded PatternLayout and does not use any context lookup. As a result, it does not rely on any
configuration that an attacker can control to execute the exploit.
```

With that said, we have just released [agent versions `7.32.4` and `6.32.4`](https://github.com/DataDog/datadog-agent/releases) that completely remove `log4j` as a dependency from `jmxfetch` to mitigate any further concerns.
